core/src/engine/task/work_stealing_queue/local_queue.hpp

```diff
@@ -14,24 +15,24 @@ class LocalQueue {
  public:
   LocalQueue() = default;
 
-  size_t Size() const { return tail_.load() - head_.load(); }
+  std::size_t Size() const noexcept { return tail_.load() - head_.load(); }
```

Comment:
In general, it can return an overflown size. Do you want to add assert ?

Reply:
Then do something like

```cpp
static_cast<std::size_t>(std::max(static_cast<std::ptrdiff_t>(...), std::ptrdiff_t{0}))
```

Otherwise this method is useless